### ISSUE TITLE
"invalid cross-device link" when moving files

### ISSUE THREAD

**Issue by UnsolvedCypher** [crystal-lang/crystal#9308]
When moving a file between filesystems using either `FileUtils.mv` or `File.rename`, I get an `invalid cross-device link` error. Issue #7777 seems to imply that this behavior wasn't the same previously. Issue #3461 was closed because this behavior is delegated to the OS. That is true, but there should be some way for a Crystal program to move/rename files independent of their location. As a workaround, I'm copying the file and then deleting the original, but that seems unidiomatic.

I'm on Linux with Crystal 0.34.0

**Comment by oprypin** [comment]
Yeah, FileUtils.mv *really* should support this.

**Comment by RX14** [comment]
`File.rename` should behave this way, and not transparently copy, but there should be some kind of `File.copy` or `File.mv` which transparently falls back to a copy+delete when rename fails.

Adding functionality to `FileUtils` which is not available outside `FileUtils` would be the wrong approach after #9175

**Comment by Blacksmoke16** [comment]
Duplicate of https://github.com/crystal-lang/crystal/issues/7777?

**Comment by UnsolvedCypher** [comment]
@Blacksmoke16 Yes, I think I misread that issue yesterday but it looks like the same thing is being discussed. I'll close this issue.
